test/jest/default-request-executor.test.js

```diff
@@ -0,0 +1,275 @@
+      const request = { method: 'GET', startTime };
+      const response = buildMockResponse({ status: 429 });
+      const returnValue = await requestExecutor.parseResponse(request, response);
+      expect(returnValue).toEqual(response);
+    });
+
+    it('should return 429 responses if the max retries has been reached', async () => {
+      const request = {
+        method: 'GET',
+        headers: {
+          'X-Okta-Retry-Count': '2'
+        },
+      };
+      const mockResponse = buildMockResponse({
+        status: 429,
+        headers: {
+          'x-rate-limit-reset': String(new Date())
+        }
+      });
+      const returnValue = await requestExecutor.parseResponse(request, mockResponse);
+      expect(returnValue).toEqual(mockResponse);
+    });
+
+    it('should return 200 responses', async () => {
+      const request = { method: 'GET' };
+      const response = buildMockResponse({ status: 200 });
+      const returnValue = await requestExecutor.parseResponse(request, response);
+      expect(returnValue).toEqual(response);
+    });
+
+    it('should return 401 responses', async () => {
+      const request = { method: 'GET' };
+      const response = buildMockResponse({ status: 401 });
+      const returnValue = await requestExecutor.parseResponse(request, response);
+      expect(returnValue).toEqual(response);
+    });
+
+    it('should return 500 responses', async () => {
+      const request = { method: 'GET' };
+      const response = buildMockResponse({ status: 500 });
+      const returnValue = await requestExecutor.parseResponse(request, response);
+      expect(returnValue).toEqual(response);
+    });
+  });
+
+  describe('requestHasTimedOut', () => {
+
+    it('should return true if the request has not timed out', () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const mockRequest = {
+        startTime: new Date()
+      };
+      const result = requestExecutor.requestHasTimedOut(mockRequest);
+      expect(result).toBe(false);
+    });
+
+    it('should return false if the request has timed out', () => {
+      const requestExecutor = new DefaultRequestExecutor({
+        requestTimeout: 1000
+      });
+      const mockRequest = {
+        startTime: new Date(new Date().getTime() - 2000)
+      };
+      const result = requestExecutor.requestHasTimedOut(mockRequest);
+      expect(result).toBe(true);
+    });
+  });
+
+  describe('retryRequest', () => {
+
+    it('should build a new request and send it to fetch', async () => {
+      const requestExecutor = new DefaultRequestExecutor();
+      const now = new Date();
+      const retryAfter = new Date(now.getTime() + (1000 * 60)); // one minute in the future
+      const mockRequest = { method: 'GET' };
+      const mockNewRequest = {};
+      requestExecutor.fetch = jest.fn();
+      requestExecutor.getRetryDelayMs = jest.fn().mockImplementation(() => 1);
```

Comment:
It's better to mock out timers with [jest's fake timers](http://jestjs.io/docs/en/timer-mocks.html) - there might be an issue with that + promises, but should try it out.